### PR TITLE
Frees up disk_vols when volume creation failed

### PR DESCRIPTION
**Pull request by scw00**
Slight optimizing.

### CODE REVIEW

**Comment by scw00** [comment]
[approve ci]

**Comment by scw00** [comment]
[approve ci]

File: iocore/cache/Cache.cc [hunk 1]
```diff
@@ -2865,6 +2865,7 @@ cplist_reconfigure()
         new_cp->disk_vols = (DiskVol **)ats_malloc(gndisks * sizeof(DiskVol *));
```
Comment (scw00) [inline iocore/cache/Cache.cc]: Should free up this memory.

**Comment by scw00** [comment]
[approve ci autest]

File: iocore/cache/Cache.cc [hunk 2]
```diff
@@ -2865,6 +2865,7 @@ cplist_reconfigure()
...
         memset(new_cp->disk_vols, 0, gndisks * sizeof(DiskVol *));
         if (create_volume(config_vol->number, size_in_blocks, config_vol->scheme, new_cp)) {
+          ats_free(new_cp->disk_vols);
```
Comment (SolidWallOfCode) [inline iocore/cache/Cache.cc]: Need to null it out then, and not leave a dangling pointer.

Comment (scw00) [reply iocore/cache/Cache.cc]: updated as requests

**Pull request merged**